byway/src/main/webapp/home.js

```diff
@@ -96,23 +126,76 @@ function initMap(start, end, route, keyString) {
   const tripMap = new google.maps.Map(document.getElementById('map-' + keyString), mapOptions);
   placesService = new google.maps.places.PlacesService(tripMap);
   directionsRenderer.setMap(tripMap);
-  calcRoute(directionsService, directionsRenderer, start, end, waypoints);
+  await setRoute(directionsService, directionsRenderer, start, end, waypoints);
+}
+
+/**
+ * Sets directions on map
+ * @param {DirectionsService} directionsService
+ * @param {DirectionsRenderer} directionsRenderer
+ * @param {String} start placeId as string
+ * @param {String} end placeId as string
+ * @param {Array} [waypoints] array of waypoint objects
+ */
+async function setRoute(directionsService, directionsRenderer, start, end, waypoints){
+  for(let i = 0; i<10; i++){
+    try{
+      let result = await getDirections(directionsService, start, end, waypoints)
+      directionsRenderer.setDirections(result);
+      return;
+    }catch(error){
+      if (error.status === google.maps.DirectionsStatus.OVER_QUERY_LIMIT){
+        await delayPromise(1000);
+      }
+    }
+  }   
```

Comment:
If you run out of tries, it would be good to at least log an error, if not throw an exception.